plugins/trace_api_plugin/trace_api_plugin.cpp

```diff
@@ -0,0 +1,229 @@
+         _log.log(fc::log_message( context, detail_string ));
+      }
+   }
+}
+
+namespace eosio {
+
+/**
+ * A common source for information shared between the extraction process and the RPC process
+ */
+struct trace_api_common_impl {
+   static void set_program_options(appbase::options_description& cli, appbase::options_description& cfg) {
+      auto cfg_options = cfg.add_options();
+      cfg_options("trace-dir", bpo::value<bfs::path>()->default_value("traces"),
+                  "the location of the trace directory (absolute path or relative to application data dir)");
+      cfg_options("slice-stride", bpo::value<uint32_t>()->default_value(10'000),
+                  "the number of blocks each \"slice\" of trace data will contain on the filesystem");
+   }
+
+   void plugin_initialize(const appbase::variables_map& options) {
+      auto dir_option = options.at("trace-dir").as<bfs::path>();
+      if (dir_option.is_relative())
+         trace_dir = app().data_dir() / dir_option;
+      else
+         trace_dir = dir_option;
+
+      slice_stride = options.at("slice-stride").as<uint32_t>();
+   }
+
+   // common configuration paramters
+   boost::filesystem::path trace_dir;
+   uint32_t slice_stride = 0;
+};
+
+/**
+ * Interface with the RPC process
+ */
+struct trace_api_rpc_plugin_impl {
+   trace_api_rpc_plugin_impl( const std::shared_ptr<trace_api_common_impl>& common )
+   :common(common) {}
+
+   static void set_program_options(appbase::options_description& cli, appbase::options_description& cfg) {
+      auto cfg_options = cfg.add_options();
+      cfg_options("trace-rpc-abi", bpo::value<vector<string>>()->composing(),
+                  "ABIs used when decoding trace RPC responses, specified as \"Key=Value\" pairs in the form <account-name>=<abi-def>\n"
+                  "Where <abi-def> can be:\n"
+                  "   a valid JSON-encoded ABI as a string\n"
+                  "   an absolute path to a file containing a valid JSON-encoded ABI\n"
+                  "   a relative path from `data-dir` to a file containing a valid JSON-encoded ABI\n"
+                  );
+   }
+
+   void plugin_initialize(const appbase::variables_map& options) {
+      data_handler = std::make_shared<abi_data_handler>([](const exception_with_context& e){
+         log_exception(e, fc::log_level::debug);
+      });
+
+      if( options.count("trace-rpc-abi") ) {
+         const std::vector<std::string> key_value_pairs = options["trace-rpc-abi"].as<std::vector<std::string>>();
+         for (const auto& entry : key_value_pairs) {
+            try {
+               auto kv = parse_kv_pairs(entry);
+               auto account = chain::name(kv.first);
+               auto abi = abi_def_from_file_or_str(kv.second);
+               data_handler->add_abi(account, abi);
+            } catch (...) {
+               elog("Malformed trace-rpc-abi provider: \"${val}\", ignoring!", ("val", entry));
```

Comment:
I don't think this should be ignored. I think log and throw, so they don't run with malformed abi because they didn't notice.